rhino/src/main/java/org/mozilla/javascript/ast/ScriptNode.java

```diff
@@ -39,6 +39,16 @@ public class ScriptNode extends Scope {
     private int tempNumber = 0;
     private boolean inStrictMode;
 
+    public boolean isMethodDefinition() {
+        return false;
+    }
+
+    public void setMethodDefinition(boolean methodDefinition) {
+        isMethodDefinition = methodDefinition;
+    }
+
+    protected boolean isMethodDefinition;
```

Comment:
Can we make it `private` if we remove the override of the getter in `FunctionNode`?